linera-core/src/client/mod.rs

```diff
@@ -260,33 +260,28 @@ impl<Env: Environment> Client<Env> {
     async fn fetch_chain_info(
         &self,
         chain_id: ChainId,
-        validators: &[RemoteNode<impl ValidatorNode>],
-    ) -> Result<Box<ChainInfo>, LocalNodeError> {
+        validators: &[RemoteNode<Env::ValidatorNode>],
+    ) -> Result<Box<ChainInfo>, ChainClientError> {
         match self.local_node.chain_info(chain_id).await {
             Ok(info) => Ok(info),
             Err(LocalNodeError::BlobsNotFound(blob_ids)) => {
+                // Make sure the admin chain is up to date.
+                self.synchronize_chain_state(self.admin_id).await?;
```

Comment:
This is potentially a performance regression. Is there a "quick" way to verify if we're up-to-date? Can we optimistically try to skip synchronization with admin chain and somehow detect it?